Makefile

```diff
@@ -1,6 +1,17 @@
 ifeq ($(OS),Windows_NT)
-    # Fail if this is a windows platform
-    $(error Windows is not supported by this makefile, please use appropriate .mak file)
+    # Need to figure out if Cygwin/Mingw is installed
+    SYS := $(shell gcc -dumpmachine)
+    ifeq ($(findstring cygwin, $(SYS)),cygwin)
+      PLATFORM = cygwin
+    endif
+    ifeq ($(findstring mingw, $(SYS)),mingw)
+      ifeq ($(PROCESSOR_ARCHITECTURE),x86)
+        PLATFORM = mingw32
+      endif
+      ifeq ($(PROCESSOR_ARCHITECTURE),AMD64)
+        PLATFORM = mingw64
+      endif
+    endif
```

Comment:
It seems that the value of `$(PROCESSOR_ARCHITECTURE)` is depended on `make.exe`.
If 64-bit `make.exe` is used, `PLATFORM` is set to `mingw64` regardless of gcc.
Maybe the following is better?
```
ifeq ($(findstring mingw32, $(SYS)),mingw32)
  PLATFORM = mingw32
endif
ifeq ($(findstring mingw64, $(SYS)),mingw64)
  PLATFORM = mingw64
endif
```